From c4c9a8d1dc5cd0ecc59471c789bc0a4555b79afa Mon Sep 17 00:00:00 2001
From: Mike Swanson <mikeonthecomputer@gmail.com>
Date: Wed, 7 Oct 2015 21:04:02 -0700
Subject: [PATCH 3/3] Properly ravel with empty lines in the commit message.

These would not round-trip before, since the blank lines got squashed.
---
 git-weave   |  2 ++
 tests/bs.fi | 12 +++++++++++-
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/git-weave b/git-weave
index a6c278b..e21e886 100755
--- a/git-weave
+++ b/git-weave
@@ -286,6 +286,8 @@ def git_ravel(indir, outdir, quiet=False):
                     if line[0].isspace():
                         if line.startswith(" " * 4):
                             line = line[4:]
+                            if line == "\n":
+                                body_latch = False
                             # Old-school byte stuffing.
                             if line.startswith("."):
                                 line = "." + line
diff --git a/tests/bs.fi b/tests/bs.fi
index 36cb2f8..6201a4e 100644
--- a/tests/bs.fi
+++ b/tests/bs.fi
@@ -18106,9 +18106,19 @@ Be explicit about releasing to freshmeat.
 from :159
 M 100644 :160 bs.spec
 
-reset refs/heads/master
+commit refs/heads/master
+mark :162
+author Mike Swanson <mikeonthecomputer@gmail.com> 1444272995 -0700
+committer Mike Swanson <mikeonthecomputer@gmail.com> 1444272995 -0700
+data 86
+Make a commit with a summary line,
+
+followed by a descriptor line after two newlines.
 from :161
 
+reset refs/heads/master
+from :162
+
 tag 2.8
 from :155
 tagger Eric S. Raymond <esr@thyrsus.com> 1287484438 -0400
-- 
2.6.1

